### PR TITLE
Feat:Collections in contentful

### PR DESCRIPTION
**Pull request by pKorsholm**
**What**
- add support for collections in Contentful 

**How**
- Add and emit new events in `ProductCollectionService` *See comments in the code for the specific implementations*
  - `CREATED: "product-collection.created"`,
  - `UPDATED: "product-collection.updated"`,
  - `DELETED: "product-collection.deleted"`,
- Create appropriate subscribers in `medusa-plugin-contentful`
- Update `product.created` and `product.updated` handlers
  - now updates the reference to the collection and the old field (to preserve backwards compatibility)
 
 *See comments in the code for the specific implementations*

**Testing**
- Expanded the unit test suite of `medusa-plugin-contentful` to include operations on collections and product creation/update

### CODE REVIEW

File: packages/medusa-plugin-contentful/src/services/contentful.js [hunk 1]
```diff
@@ -594,124 +588,217 @@ class ContentfulService extends BaseService {
...
+      {
+        fields,
       }
+    )
+
+    return result
+  }
 
-      variantEntry.fields = variantEntryFields
+  async updateProductCollectionInContentful(data) {
```
Comment (pKorsholm) [inline packages/medusa-plugin-contentful/src/services/contentful.js:675]: Update Product Collection when a collection is updated in Medusa

File: packages/medusa-plugin-contentful/src/services/contentful.js [hunk 2]
```diff
@@ -594,124 +588,217 @@ class ContentfulService extends BaseService {
...
+
+    const updatedEntry = await variantEntry.update()
+    const publishedEntry = await updatedEntry.publish()
+
+    await this.addIgnore_(variant.id, "medusa")
+
+    return publishedEntry
+  }
+
+  async createProductCollectionInContentful(data) {
```
Comment (pKorsholm) [inline packages/medusa-plugin-contentful/src/services/contentful.js:639]: Create Product Collection when a collection is created in Medusa

File: packages/medusa-plugin-contentful/src/services/contentful.js [hunk 3]
```diff
@@ -142,165 +137,165 @@ class ContentfulService extends BaseService {
   }
 
   async createProductInContentful(product) {
```
Comment (pKorsholm) [inline packages/medusa-plugin-contentful/src/services/contentful.js:139]: Handle product creation with collection in Medusa

File: packages/medusa-plugin-contentful/src/services/contentful.js [hunk 4]
```diff
@@ -366,53 +357,49 @@ class ContentfulService extends BaseService {
...
+    await this.addIgnore_(data.id, "medusa")
 
-      return publishedEntry
-    } catch (error) {
-      throw error
-    }
+    return publishedEntry
   }
 
   async updateProductInContentful(data) {
```
Comment (pKorsholm) [inline packages/medusa-plugin-contentful/src/services/contentful.js:405]: Handle product update with collection in Medusa

**Comment by srindom** [comment]
We should make it possible to opt-out of collections sync - we have users that use the plugin right now without collections; we should allow them to continue using it like that :)

File: packages/medusa-plugin-contentful/src/services/contentful.js [hunk 5]
```diff
@@ -594,124 +588,217 @@ class ContentfulService extends BaseService {
...
+    const publishedEntry = await updatedEntry.publish()
+
+    await this.addIgnore_(data.id, "medusa")
+
+    return publishedEntry
   }
 
-  async archiveProductVariantInContentful(variant) {
-    let variantEntity
+  async archiveProductCollectionInContentful(collection) {
```
[the remark this answers is not in the document]
Comment (vuuzle) [reply packages/medusa-plugin-contentful/src/services/contentful.js:728]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/src/services/contentful.js (hunk 1 above)
Comment (vuuzle) [reply packages/medusa-plugin-contentful/src/services/contentful.js:675]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/src/services/contentful.js (hunk 2 above)
Comment (vuuzle) [reply packages/medusa-plugin-contentful/src/services/contentful.js:639]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

File: packages/medusa-plugin-contentful/src/services/contentful.js (hunk 3 above)
Comment (vuuzle) [reply packages/medusa-plugin-contentful/src/services/contentful.js:139]: https://vuuzletvph.com/heads-up-vuco-bounty-programs-are-coming-your-way/

**Pull request closed without merging**